File: src/main/java/dev/sky_lock/pocketlifevehicle/command/GiveCommand.java
package dev.sky_lock.pocketlifevehicle.command;

import dev.sky_lock.pocketlifevehicle.PLVehicle;
import dev.sky_lock.pocketlifevehicle.car.ModelList;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class GiveCommand implements ICommand, IAdminCommand {

    @Override
    public void execute(CommandSender sender, Command cmd, String[] args) {
        Player player = (Player) sender;
        if (args.length < 3) {
            player.sendMessage(PLVehicle.PREFIX + ChatColor.RED + "引数が足りません");
            return;
        }
        String name = args[1];
        Player target = Bukkit.getPlayer(name);
        if (target == null) {
            player.sendMessage(PLVehicle.PREFIX + ChatColor.RED + "プレイヤーが見つかりませんでした");
            return;
        }
        String id = args[2];
        ModelList.of(id).map(model -> {
            player.getInventory().addItem(model.getItemStack());
            player.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + name + "に" + id + "を与えました");
            target.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + "車を受け取りました");
            return model;
        }).orElseGet(() -> {
            player.sendMessage(PLVehicle.PREFIX + ChatColor.RED + "車種が見つかりませんでした");
            return null;
        });
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/command/GiveCommand.kt
package dev.sky_lock.pocketlifevehicle.command

import dev.sky_lock.pocketlifevehicle.extension.chat.sendVehiclePrefixedErrorMessage
import dev.sky_lock.pocketlifevehicle.extension.chat.sendVehiclePrefixedSuccessMessage
import dev.sky_lock.pocketlifevehicle.vehicle.ModelRegistry
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class GiveCommand : ICommand, IAdminCommand {

    override fun execute(sender: CommandSender, cmd: Command, args: Array<String>) {
        val player = sender as Player
        if (args.size < 2) {
            player.sendVehiclePrefixedErrorMessage( "引数が足りません")
            return
        }
        val id = args[1]
        val model = ModelRegistry.findById(id)
        if (model == null) {
            player.sendVehiclePrefixedErrorMessage( "モデルが見つかりませんでした")
            return
        }
        if (model.flag.eventOnly) {
            player.sendVehiclePrefixedErrorMessage( "イベント専用車両にはspawnコマンドを使って下さい")
            return
        }
        if (args.size == 2) {
            player.inventory.addItem(model.itemStack)
            player.sendVehiclePrefixedSuccessMessage( id + "を取得しました")
            return
        }
        val name = args[2]
        val target = Bukkit.getPlayer(name)
        if (target == null) {
            player.sendVehiclePrefixedErrorMessage( "プレイヤーが見つかりませんでした")
            return
        }
        target.inventory.addItem(model.itemStack)
        target.sendVehiclePrefixedSuccessMessage( "乗り物を受け取りました")
        player.sendVehiclePrefixedSuccessMessage( name + "に" + id + "を与えました")
    }
}